docs/en/advanced_guides/add_dataset.md

```diff
@@ -58,99 +99,264 @@ We use `RepeatDataset` as wrapper to repeat the dataset.
-```
+````
+You may concatenate two `dataset`.
+
+  ```python
+    dataset_train_1 = dict(
+        type=dataset_type,
+        data_root=data_root,
+        data_prefix=dict(img_path='img_dir', seg_map_path='ann_dir_1'),
+        ann_file='ann_file_1.txt',
+        pipeline=train_pipeline)
+    dataset_train_2 = dict(
+        type=dataset_type,
+        data_root=data_root,
+        data_prefix=dict(img_path='img_dir', seg_map_path='ann_dir_2'),
+        ann_file='ann_file_2.txt',
+        pipeline=train_pipeline)
+    train_dataloader = dict(
+        batch_size=4,
+        num_workers=4,
+        persistent_workers=True,
+        sampler=dict(type='InfiniteSampler', shuffle=True),
+        dataset=dict(type='ConcatDataset', datasets=[dataset_train_1, dataset_train_2]))
+  ```
+
+  In this case, `ann_dir_1` and `ann_dir_2` are corresponding to `ann_file_1.txt` and `ann_file_2.txt`.
+````
+
+### Multi-image Mix Dataset
+
+We use `MultiImageMixDataset` as a wrapper to mix images from multiple datasets.
+`MultiImageMixDataset` can be used by multiple images mixed data augmentation
+like mosaic and mixup.
 
-A more complex example that repeats `Dataset_A` and `Dataset_B` by N and M times, respectively, and then concatenates the repeated datasets is as the following.
+An example of using `MultiImageMixDataset` with `Mosaic` data augmentation:
 
 ```python
-dataset_A_train = dict(
-    type='RepeatDataset',
-    times=N,
-    dataset=dict(
-        type='Dataset_A',
-        ...
-        pipeline=train_pipeline
-    )
-)
-dataset_A_val = dict(
-    ...
-    pipeline=test_pipeline
-)
-dataset_A_test = dict(
-    ...
-    pipeline=test_pipeline
-)
-dataset_B_train = dict(
-    type='RepeatDataset',
-    times=M,
+train_pipeline = [
+    dict(type='LoadImageFromFile'),
+    dict(type='LoadAnnotations'),
+    dict(type='RandomMosaic', prob=1),
+    dict(
+        type='RandomResize',
+        scale=(2048, 512),
+        ratio_range=(0.5, 2.0),
+        keep_ratio=True),
+    dict(type='RandomCrop', crop_size=crop_size, cat_max_ratio=0.75),
+    dict(type='RandomFlip', prob=0.5),
+    dict(type='PhotoMetricDistortion'),
+    dict(type='PackSegInputs')
+]
+
+train_dataloader = dict(
+    batch_size=4,
+    num_workers=4,
+    persistent_workers=True,
+    sampler=dict(type='InfiniteSampler', shuffle=True),
     dataset=dict(
-        type='Dataset_B',
-        ...
-        pipeline=train_pipeline
-    )
+        type='MultiImageMixDataset',
+        dataset=dict(
+            classes=classes,
+            palette=palette,
+            type=dataset_type,
+            reduce_zero_label=False,
+            data_prefix=dict(img_path=data_root + "images/train",
+                             seg_map_path=data_root + "annotations/train"),
+            pipeline=[
+                dict(type='LoadImageFromFile'),
+                dict(type='LoadAnnotations'),
+            ]
+        ),
+        pipeline=train_pipeline)
```

Comment:
what is the difference between `pipeline` in line 185 and `pipeline` in line 190